apps/explorer/src/ui/ExpandableList.tsx

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { ChevronDown12 } from '@mysten/icons';
+import { type ReactNode, useMemo, useState } from 'react';
+
+import { Text } from './Text';
+
+interface ExpandableListProps {
+    children: ReactNode[];
+    defaultItemsToShow?: number;
+}
+
+export function ExpandableList({
+    children,
+    defaultItemsToShow = 3,
+}: ExpandableListProps) {
+    const [showAll, setShowAll] = useState(false);
+    const items = useMemo(
+        () => (showAll ? children : children.slice(0, defaultItemsToShow)),
+        [showAll, children, defaultItemsToShow]
+    );
+
+    const handleShowAllClick = () =>
+        setShowAll((prevShowAll: boolean) => !prevShowAll);
+
+    return (
+        <div className="flex flex-col gap-4">
+            {items.map((item, index) => (
+                <div key={index}>{item}</div>
+            ))}
+            {children.length > defaultItemsToShow && (
```

Comment:
`children` should be treated like an opaque type because it's treated a little special syntactically. If you want to do something like this, just take a dedicated prop for it (e.g. `list`). It'll be less error-prone in usage.